### PR TITLE
Fix CI and cargo build errors

### PR DESCRIPTION
**Pull request by wenyuzhao**
* Fix GitHub actions so it runs on ubuntu-20.04 and can pass all the v8 unit tests (with mmtk nogc). But it still pins a relatively old v8 commit.
* Fix all the cargo build errors
* Update rust-toolchain to match mmtk-core
* Update mmtk-v8 version to v0.18.0

---

Note that v8 with mmtk-v8 still can't build on ubuntu-22.04, until we move forward the pinned v8 commit to a version that supports 22.04.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   test:
-    runs-on: [self-hosted, linux]
+    runs-on: ubuntu-20.04
     if: contains(github.event.pull_request.labels.*.name, 'PR-approved')
```
Comment (qinsoon) [inline .github/workflows/ci.yml:12]: This line can be removed. It was used to make sure we do not run arbitrary code on our CI machines. As we run on Github runner now, we don't need it anymore.

**Comment by qinsoon** [comment]
Thanks for the PR!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (wenyuzhao) [reply .github/workflows/ci.yml]: Done

File: .github/scripts/build-mmtk.sh [hunk 2]
```diff
@@ -3,5 +3,10 @@ set -xe
 . $(dirname "$0")/common.sh
 
 # simply build mmtk-v8 with nogc
-cd $THE_ROOT/mmtk
-rustup run $RUSTUP_TOOLCHAIN cargo build --features nogc
+rustup toolchain install $RUSTUP_TOOLCHAIN
```
Comment (caizixian) [inline .github/scripts/build-mmtk.sh:6]: Perhaps change entire thing to 
```suggestion
rustup toolchain install $RUSTUP_TOOLCHAIN --target x86_64-unknown-linux-gnu  --component clippy rustfmt
```

**Review by caizixian — APPROVED** [review]

File: .github/scripts/build-mmtk.sh (hunk 2 above)
Comment (wenyuzhao) [reply .github/scripts/build-mmtk.sh]: Done

**Pull request merged**